crates/chia-bls/src/cached_bls.rs

```diff
@@ -44,13 +43,13 @@ impl BlsCache {
         self.cache.is_empty()
     }
 
-    pub fn aggregate_verify(
+    pub fn aggregate_verify<Pk: Borrow<PublicKey>, Msg: AsRef<[u8]>>(
         &mut self,
-        pks: impl IntoIterator<Item = impl Borrow<PublicKey>>,
-        msgs: impl IntoIterator<Item = impl AsRef<[u8]>>,
+        pks_msgs: impl IntoIterator<Item = (Pk, Msg)>,
         sig: &Signature,
-    ) -> bool {
-        let iter = pks.into_iter().zip(msgs).map(|(pk, msg)| -> GTElement {
+    ) -> (bool, HashMap<[u8; 32], GTElement>) {
```

Comment:
The python counterpart has a function that returns the content of the cache, rather than making `aggregate_verify()` return all the entries that were added. I can imagine a future, when we've replaced the worker processes with threads, where the BLS cache is truly shared between the block validation threads and the TX validation threads. At that point we no longer need to return the cached items.

Right now, asking for the items and inserting the items is done in the python binding. What do you think about moving those functions into the proper rust type instead?

Also, I think returning a `Vec<([u8; 32], GTElement)>` would be quite a lot cheaper too.